preview-env/create/action.yml

```diff
@@ -133,3 +133,23 @@ runs:
       env_url: ${{ inputs.app_url }}
       ref: ${{ inputs.revision }}
       env: ${{ inputs.app_name }}
+  - name: Create deployment status artifact
+    if: always()
+    shell: bash
+    env:
+      APP_NAME: ${{ inputs.app_name }}
+      APP_URL: ${{ inputs.app_url }}
+      ARGOCD_SERVER: ${{ inputs.argocd_server }}
+      REPO: ${{ github.repository }}
+      RUN_ID: ${{ github.run_id }}
+    run: |
+      template=$(envsubst < ${{ github.action_path }}/comment_template_${{ github.action_status }}.md)
```

Comment:
I would put the templates in a dedicated folder like:
- templates/comment-failure.md
- templates/comment-success.md
- templates/comment-cancelled.md